### DISCUSSION TITLE
Is the traffic between filer and Harvest encrypted?

### DISCUSSION THREAD

**Discussion by jmg011** [NetApp/harvest#1639] [Q&A] [answered]
Is the traffic between filer and Harvest encrypted?

**Comment by rahulguptajss** [comment]
@jmg011  By default,  Harvest uses encrypted communication for filer. However this can be turned off by setting `use_insecure_tls` to `true` in harvest configuration. More information [here](https://netapp.github.io/harvest/22.11/configure-harvest-basic/) 

![image](https://user-images.githubusercontent.com/25551691/208808486-3ac141c4-812b-41c9-9825-7ac4f76bc62a.png)

**Reply by jmg011** [reply]
Thank you @rahulguptajss

**Reply by cgrinds** [reply, marked as answer]
Even with `use_insecure_tls`, HTTPS is used as the transport and is encrypted. Specifically setting `use_insecure_tls` to true enables Go's `InsecureSkipVerify` which controls whether a client (Harvest in this case) verifies the server's certificate chain and host name (ONTAP). 

If `InsecureSkipVerify` is true, crypto/tls accepts any certificate presented by the server and any host name in that certificate but the transport is still encrypted.
